addons/pos_self_order/models/pos_session.py

```diff
@@ -10,7 +10,8 @@ class PosSession(models.Model):
     @api.model_create_multi
     def create(self, vals_list):
         sessions = super(PosSession, self).create(vals_list)
-        sessions = self._create_pos_self_sessions_sequence(sessions)
+        if sessions.config_id.self_ordering_mode != 'nothing':
+            sessions = self._create_pos_self_sessions_sequence(sessions)
```

Comment:
Is this really required for the fix to work ? It makes sense not to create a useless sequence but if it's not required let's keep the changes minimal